ensure MCBP session closure when topology changes

avsej opened the pull request.

davidkelly reviewed (approved):
LGTM

brett19 commented:
I wonder if we should modify certain things like mcbp_session to be non-copyable?

avsej commented on couchbase/bucket.hxx:
```diff
@@ -213,7 +214,7 @@ class bucket : public std::enable_shared_from_this<bucket>
             new_session = std::make_shared<io::mcbp_session>(client_id_, ctx_, origin_, name_, known_features_);
         }
-        new_session->bootstrap([self = shared_from_this(), new_session, h = std::forward<Handler>(handler)](
+        new_session->bootstrap([self = shared_from_this(), new_session = std::move(new_session), h = std::forward<Handler>(handler)](
```
yes, it is not necessary here, but above it is the last usage of the pointer, so we can safely move it into the closure

avsej commented:
@brett19 I've updated the patch, and removed questionable `std::move`. Now the only move left in the place, where we know for sure the pointer is not used anymore
